### ISSUE TITLE
Font: can't fix font

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#116]
![image](https://cloud.githubusercontent.com/assets/4609956/23705433/1128d6da-040a-11e7-94d4-0b854e3a20e5.png)

vitaMalacologica.14.1_2016 Bank Neubert Enidae Iran.pdf

**Comment by gsautter** [comment]
"Edit Font" doesn't quite cut it, I take it? What happens if you try to open that dialog?

And as there recently were a few changes in how words are linked to fonts, when exactly did you decode this one, in particular with which build? Any changes with the latest build that I put out on Monday?

**Comment by myrmoteras** [comment]
The build most recent
Files processed this afternoon

Get Outlook for Android<https://aka.ms/ghei36>


________________________________

**Comment by myrmoteras** [comment]
The dialog does not offer this letter to be correct

Get Outlook for Android<https://aka.ms/ghei36>


________________________________

**Comment by gsautter** [comment]
OK, that sounds weird ... could you put the original PDF here for me to download and test?

**Comment by myrmoteras** [comment]
The file is under the name vitaMalacologica.14.1_2016 Bank Neubert Enidae Iran.pdf in the SRS store.

**Comment by gsautter** [comment]
Tip for PDFs that copy & paste nicely (e.g. to some text editor): Try decoding with option "<Render Glyphs Only>", as that does render the characters for later correction, but doesn't question the Unicode transcription.
Always a good idea to start with "<Render Glyphs Only>" anyway, and only if you see characters are scrambled, resort to the fully blown font decoding under "Default". As it omits most of the decoding effort, "<Render Glyphs Only>" is on top of that faster than "Default", and by some margin so.

**Comment by gsautter** [comment]
Figured it out ... and what a weird one.

In PDFs, fonts can be derived from other fonts (dubbed "base fonts" in that role), borrowing the glyphs (character images), but might well use a different encoding than the base font, e.g. to eliminate &gt;255 char codes, which take up more space because they take 2 bytes to store rather than 1. So far, so fine.

Now each individual font is usually used in either _one_ role in a PDF (main font _or_ base font), but rarely ever both. However, in this very PDF, we have as many as three fonts in the dual roles.
This caused collisions with the font names, which are the means of binding each word to a specific font.
Those collisions, in turn, incurred a detachment of words from the very fonts that translate their character codes into actual Unicode characters. With that link broken, GGI has no means of figuring out in which words a given character code from a given font is used, aka which words are affected by a change to the Unicode transcription of a given character code. And exactly the latter is required when writing font edits through to word strings and page images ...

Long story short: the PDF decoder prevents these collisions now, so such mix-ups cease to happen.

**Comment by gsautter** [comment]
I've fixed the document fonts now, the repaired version is on the server.
UUID is FFD6447AFFE8192CB02DFFAC44309D77

There's also an update out with the aforementioned fix.
